livekit/src/room/mod.rs

```diff
@@ -409,6 +409,8 @@ impl Room {
             room_task: Default::default(),
         });
 
+        inner.set_sid(room_info.sid.try_into().unwrap());
```

Comment:
It is OK because we're still using the Protocol 9 https://github.com/livekit/rust-sdks/blob/8b276f9d4b98437a139e1cbe41cfda1332ce1120/livekit-api/src/signal_client/mod.rs#L47.

Otherwise we should keep in mind that the `async sid `method only resolves once it has a valid SID from the server